src/coreclr/jit/emitxarch.h

```diff
@@ -43,7 +43,7 @@ inline static bool isHighGPReg(regNumber reg)
 #ifdef TARGET_AMD64
     // TODO-apx: the definition here is incorrect, we will need to revisit this after we extend the register definition.
     //           for now, we can simply use REX2 as REX.
-    return ((reg >= REG_R8) && (reg <= REG_R15));
+    return ((reg >= REG_R16) && (reg <= REG_R23));
```

Comment:
I haven't looked at all the users, but I'm surprised this wouldn't cause diffs. "high GPR" means something for x64 versus x86. Maybe there should be a separate `isApxEgpr()`?

Reply:
This was a method added by Ruihan specifically for APX/Rex2/eGPR(https://github.com/dotnet/runtime/commit/3410c763587fe2dca4d1dc5888f91708a6fff816#diff-782ed843d790c8cf94cba03c0d408a37c64cdba7832dbb5a560f76979355bdd2R41-R51). He initially just used `REG_R8` and `REG_R15` since we didn't have REG_R16 and above. This code would have been inactive till now

Reply:
This comes into play only if/when register allocator selects eGPR and emitter has to encode REX2 for eGPR

Reply:
Shouldn't the range for non-APX cases still be `R8~R15`? I mean we can have it extend to `R8~R23` because we know there will not be any registers we will see between `R16~R23` for non-APX, but it should at least start from `R8` rather than `R16`, right?